File: frontend/app/stores/awards.ts
export type AwardCyclePhase = 'suggesting' | 'voting' | 'decided'

export interface AwardVoteTally {
  candidate_user_id: string
  vote_count: number
}

export interface AwardCycle {
  id: string
  month: string
  phase: AwardCyclePhase
  drawn_category_title: string | null
  drawn_category_emoji: string | null
  drawn_category_suggested_by_id: string | null
  duty_master_winner_id: string | null
  duty_master_win_count: number | null
  community_award_winner_id: string | null
  community_award_vote_count: number | null
  community_award_vetoed: boolean
  finalized_at: string | null
}

export interface AwardCurrentState extends AwardCycle {
  my_suggestion_submitted: boolean
  my_vote_candidate_id: string | null
  votes: AwardVoteTally[]
}

export interface AwardSummary {
  current: AwardCurrentState | null
  latest_decided: AwardCycle | null
}

export interface MemberAwardBadge {
  month: string
  kind: 'duty_master' | 'community'
  title: string | null
  emoji: string | null
}

export interface MemberAwardHistory {
  user_id: string
  badges: MemberAwardBadge[]
}

export const useAwardsStore = defineStore('awards', {
  state: () => ({
    summary: null as AwardSummary | null,
  }),
  actions: {
    async fetchSummary() {
      this.summary = await $fetch<AwardSummary>('/api/awards/summary')
      return this.summary
    },

    async suggest(title: string, emoji: string) {
      await $fetch('/api/awards/suggestions', { method: 'POST', body: { title, emoji } })
      await this.fetchSummary()
    },

    async vote(cycleId: string, candidateUserId: string) {
      await $fetch(`/api/awards/cycles/${cycleId}/vote`, {
        method: 'PUT',
        body: { candidate_user_id: candidateUserId },
      })
      await this.fetchSummary()
    },

    async veto(cycleId: string, reason?: string) {
      await $fetch(`/api/awards/cycles/${cycleId}/veto`, { method: 'POST', body: { reason: reason ?? null } })
      await this.fetchSummary()
    },

    async fetchMemberHistory(userId: string) {
      return await $fetch<MemberAwardHistory>(`/api/awards/members/${userId}/history`)
    },
  },
})

File: app/schemas/awards.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel, Field


class AwardVoteTallyOut(BaseModel):
    candidate_user_id: uuid.UUID
    vote_count: int


class AwardCycleOut(BaseModel):
    id: uuid.UUID
    month: date
    phase: str
    drawn_category_title: str | None
    drawn_category_emoji: str | None
    drawn_category_suggested_by_id: uuid.UUID | None
    duty_master_winner_id: uuid.UUID | None
    duty_master_win_count: int | None
    community_award_winner_id: uuid.UUID | None
    community_award_vote_count: int | None
    community_award_vetoed: bool
    finalized_at: datetime | None


class AwardCurrentStateOut(AwardCycleOut):
    my_suggestion_submitted: bool
    my_vote_candidate_id: uuid.UUID | None
    votes: list[AwardVoteTallyOut]


class AwardSummaryOut(BaseModel):
    current: AwardCurrentStateOut | None
    latest_decided: AwardCycleOut | None


class AwardSuggestionCreate(BaseModel):
    title: str = Field(min_length=1, max_length=100)
    emoji: str = Field(min_length=1, max_length=32)


class AwardVoteIn(BaseModel):
    candidate_user_id: uuid.UUID


class AwardVetoIn(BaseModel):
    reason: str | None = Field(default=None, max_length=255)


class AwardTickIn(BaseModel):
    as_of: date | None = None


class MemberAwardBadgeOut(BaseModel):
    month: date
    kind: str
    title: str | None
    emoji: str | None


class MemberAwardHistoryOut(BaseModel):
    user_id: uuid.UUID
    badges: list[MemberAwardBadgeOut]
